packages/aws-cdk-lib/aws-efs/lib/efs-file-system.ts

```diff
@@ -350,6 +357,44 @@ export interface FileSystemAttributes {
   readonly fileSystemArn?: string;
 }
 
+/**
+ * Replication configuration for the file system.
+ */
+export interface ReplicationConfiguration {
```

Comment:
Since there are multiple ways to configure the destination file system in `ReplicationConfiguration`, I'd suggest making this an enum-like class with static functions to enforce stronger typing and coupling between required properties. For example:

```ts
export abstract class ReplicationConfiguration {
    
    public static destinationFileSystem(fileSystem: IFileSystem): ReplicationConfiguration {
        // configure properties and return 
    }

    public static oneZoneFileSystem(availabilityZone: string, region: string, kmsKey?: kms.IKey): ReplicationConfiguration {
        // configure properties and return 
    }

    public static regionalFileSystem(region: string, kmsKey?: kms.IKey): ReplicationConfiguration {
        // configure properties and return  
    }
}
```

This should also reduce the need for some of the validation/error-handling below which is ideal.

Reply:
Sounds good! I'll try to implement it.